uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/main.controller.js

```diff
@@ -45,6 +47,67 @@ define([
       }
     }
 
+    /**
+     * Displays a case closed success message given the provided case and client
+     * data.
+     *
+     * @param {String} caseTypeTitle - the case type title.
+     * @param {String} clientDisplayName - the display name of the case client.
+     */
+    function displayCaseClosedSuccessMessage (caseTypeTitle, clientDisplayName) {
+      var messageTemplte = 'All tasks in the {Case type} workflow for {Case client} have been completed. Good work!';
+      var message = messageTemplte.replace('{Case type}', caseTypeTitle)
+        .replace('{Case client}', clientDisplayName);
+
+      notificationService.success('Success', message);
+    }
+
+    /**
+     * Returns the parent's case title and status for the given child task id.
+     *
+     * @param {Number} taskId - the child task id.
+     * @return {Promise} resolves to the parent case title and status.
+     */
+    function getCaseDataFromTask (taskId) {
+      return taskService.get({
+        id: taskId,
+        return: [
+          'case_id.case_type_id.title',
+          'case_id.status_id.name'
+        ]
+      }).then(function (result) {
+        if (result.length === 0) {
+          return;
+        }
+
+        return {
+          caseTypeTitle: result[0]['case_id.case_type_id.title'],
```

Comment:
Not sure what `result[0]` is, could it be a bit more descriptive (stored in a separate var?)